crates/viewer/re_viewer_context/src/cache/video_stream_cache.rs

```diff
@@ -0,0 +1,555 @@
+    let query_results = store.storage_engine().cache().range(
+        &entire_timeline_query,
+        entity_path,
+        &[frame_sample_descr.clone(), codec_chunk_descr.clone()],
+    );
+    let video_chunks = query_results
+        .get_required(&frame_sample_descr)
+        .map_err(|_err| VideoStreamProcessingError::NoVideoSamplesFound)?;
+    let codec_chunks = query_results
+        .get_required(&codec_chunk_descr)
+        .map_err(|_err| VideoStreamProcessingError::MissingCodec)?;
+
+    // Translate codec by looking at the last codec.
+    // TODO(andreas): Should validate whether all codecs ever logged are the same, but it's a bit tedious.
+    let last_codec = codec_chunks
+        .last()
+        .and_then(|chunk| {
+            chunk.component_instance::<components::VideoCodec>(&codec_chunk_descr, 0, 0)
+        })
+        .ok_or(VideoStreamProcessingError::MissingCodec)?
+        .map_err(VideoStreamProcessingError::FailedReadingCodec)?;
+    let codec = match last_codec {
+        components::VideoCodec::H264 => re_video::VideoCodec::H264,
+        // components::VideoCodec::H265 => re_video::VideoCodec::H265,
+        // components::VideoCodec::VP8 => re_video::VideoCodec::Vp8,
+        // components::VideoCodec::VP9 => re_video::VideoCodec::Vp9,
+        // components::VideoCodec::AV1 => re_video::VideoCodec::Av1,
+    };
+
+    // Extract all video samples.
+    let mut video_sample_buffers = StableIndexDeque::new();
+    let mut samples = StableIndexDeque::with_capacity(video_chunks.len()); // Number of video chunks is minimum number of samples.
+    let mut gops = StableIndexDeque::new();
+
+    for chunk in video_chunks {
+        read_samples_from_chunk(
+            timeline,
+            &frame_sample_descr,
+            chunk,
+            codec,
+            &mut video_sample_buffers,
+            &mut samples,
+            &mut gops,
+        )?;
+    }
+
+    Ok((
+        re_video::VideoDataDescription {
+            codec,
+            stsd: None,                               // Only on mp4s have this.
+            coded_dimensions: None,                   // Unknown so far.
+            timescale: re_video::Timescale::IDENTITY, // We use raw rerun timestamps, so we don't have to scale time.
+            duration: None, // Streams have to be assumed to be open ended, so we don't have a duration.
+            gops,
+            samples,
+            samples_statistics: re_video::SamplesStatistics::NO_BFRAMES, // TODO(#10090): No b-frames for now.
+            mp4_tracks: std::iter::once((0, Some(re_video::TrackKind::Video))).collect(),
+        },
+        video_sample_buffers,
+    ))
+}
+
+/// Reads all samples from a chunk and appends them to a list.
+///
+/// Rejects out of order samples - new samples must have a higher timestamp than the previous ones.
+fn read_samples_from_chunk(
+    timeline: TimelineName,
+    frame_chunk_descr: &re_types::ComponentDescriptor,
```

Comment:
Do we really need to pass this by argument? It's just `VideoStream::descriptor_sample()` anyways?

Reply:
also: name